uaclient/system.py

```diff
@@ -370,14 +371,25 @@ def create_file(filename: str, mode: int = 0o644) -> None:
     os.chmod(filename, mode)
 
 
-def write_file(filename: str, content: str, mode: int = 0o644) -> None:
+def write_file(
+    filename: str, content: str, mode: Optional[int] = None
+) -> None:
     """Write content to the provided filename encoding it if necessary.
```

Comment:
Can we add some content to this docstring describing the new behavior - that it preserves ownership and permissions of an existing file unless `mode` is not-None